client/me/concierge/book/info-step.js

```diff
@@ -168,7 +168,7 @@ class InfoStep extends Component {
 							includeManualOffsets={ false }
 							name="timezone"
 							onSelect={ this.setTimezone }
-							selectedZone={ timezone }
+							selectedZone={ maybeRewriteTimezone( timezone ) }
```

Comment:
The `timezone` value comes from Redux (`state.concierge.signupForm`) and its default value is the result of `moment.tz.guess()` (see the `concierge/signup-form` reducer code), i.e., it's not rewritten. If I don't change the default value using the UI, the not-rewritten default timezone will be sent as part of the submitted form params. That's not correct.

This can be fixed by rewriting the initial value in the reducer:
```js
const reducer = ( state = maybeRewriteTimezone( moment.tz.guess() ), action ) => {
  ...
}
```

Reply:
Ok, I'll look into that. I thought @niranjan-uma-shankar said it was working there too?

Reply:
> selectedZone={ maybeRewriteTimezone( timezone ) }

This makes the rewritten timezone value show as selected in the UI but @jsnajdr  is right that it does not persist this to state, so the not-rewritten value gets submitted as form param. Not persisting to state does not have an effect on the outcome since the not-rewritten value is still a valid timezone that can be parsed- e.g in the next step of the concierge booking form, the timezone indicated in the appointment time dropdown is correct even though it uses the not-rewritten value. Still, the clean way to do it would be to persist - thanks for catching that @jsnajdr.